packages/access-api/src/routes/validate-email.js

```diff
@@ -173,7 +173,7 @@ async function authorize(req, env) {
     })
     if (confirmResult.error) {
```

Comment:
This now errors. The message is
```
error in validate-email Error: error confirming
    at authorize3 (/Users/bengo/protocol.ai/w3protocol/packages/access-api/src/routes/validate-email.js:175:13)
    at Router.fetch (/Users/bengo/protocol.ai/w3protocol/node_modules/.pnpm/@web3-storage+worker-utils@0.4.3-dev/node_modules/@web3-storage/worker-utils/src/router.js:141:15)
    ... 2 lines matching cause stack trace ...
    at Context.<anonymous> (file:///Users/bengo/protocol.ai/w3protocol/packages/access-api/test/access-authorize.test.js:124:17) {
  [cause]: InvalidAudience: Expected literal "did:web:test.web3.storage" instead got "did:mailto:dag.house:email"
      at confirm (/Users/bengo/protocol.ai/w3protocol/node_modules/.pnpm/@ucanto+server@6.1.0/node_modules/@ucanto/server/src/handler.js:54:14)
      at authorize3 (/Users/bengo/protocol.ai/w3protocol/packages/access-api/src/routes/validate-email.js:170:33)
      ... 3 lines matching cause stack trace ...
      at Context.<anonymous> (file:///Users/bengo/protocol.ai/w3protocol/packages/access-api/test/access-authorize.test.js:124:17) {
    cause: LiteralError: Expected literal "did:web:test.web3.storage" instead got "did:mailto:dag.house:email"
        at Literal.readWith (/Users/bengo/protocol.ai/w3protocol/node_modules/.pnpm/@ucanto+validator@6.1.0/node_modules/@ucanto/validator/src/schema/schema.js:1018:9)
        at Literal.read (/Users/bengo/protocol.ai/w3protocol/node_modules/.pnpm/@ucanto+validator@6.1.0/node_modules/@ucanto/validator/src/schema/schema.js:40:17)
        at confirm (/Users/bengo/protocol.ai/w3protocol/node_modules/.pnpm/@ucanto+server@6.1.0/node_modules/@ucanto/server/src/handler.js:52:35)
        at authorize3 (/Users/bengo/protocol.ai/w3protocol/packages/access-api/src/routes/validate-email.js:170:33)
        at Router.fetch (/Users/bengo/protocol.ai/w3protocol/node_modules/.pnpm/@web3-storage+worker-utils@0.4.3-dev/node_modules/@web3-storage/worker-utils/src/router.js:141:15)
        at Object.fetch (/Users/bengo/protocol.ai/w3protocol/packages/access-api/src/index.js:29:19)
        at ServiceWorkerGlobalScope.[kDispatchFetch] (/Users/bengo/protocol.ai/w3protocol/node_modules/.pnpm/@miniflare+core@2.11.0/node_modules/@miniflare/core/src/standards/event.ts:385:13)
        at Context.<anonymous> (file:///Users/bengo/protocol.ai/w3protocol/packages/access-api/test/access-authorize.test.js:124:17) {
      expect: 'did:web:test.web3.storage',
      actual: 'did:mailto:dag.house:email'
    }
  }
}
```

I think this makes sense.... the invocation has `audience` of the `did:mailto:...` but here I pass `id: env.signer.verifier`. I expected the pre-signed invocation would be addressed to us, not the `did:mailto`.

I could:
* change this call to `confirm` to pass `id: theDidMailto`
* change the `access/confirm` invocation created by `access/authorize` handler to use `audience: service`

Reply:
https://github.com/web3-storage/w3protocol/actions/runs/4421808718/jobs/7752991464#step:7:25

Reply:
@Gozala any advice on which of the above paths makes sense? or if I 'm missing something?

Reply:
> change this call to confirm to pass id: theDidMailto

Actually I can't do that. `id` must be a Verifier, and idk if there is a good Verifier when audience is `did:mailto`.

Reply:
I found a way of doing with provideAdvanced https://github.com/web3-storage/w3protocol/pull/541/commits/6045824208141c8216c6a6b6a4967b6c839e41d8#diff-dc3f2f4fda92c2be86974bff5c145103773f6c7cb68d879d52d464392702243dR160

Reply:
I'm not sure I have full context why you chose to switch to `provideAdvanced` here. Given how it's used here I would guess that regular `provide` is what you want.